core/shared/src/main/scala/org/http4s/Uri.scala

```diff
@@ -339,12 +340,16 @@ object Uri extends UriPlatform {
     private def doEquals(path: Path): Boolean =
       this.segments == path.segments && path.absolute == this.absolute && path.endsWithSlash == this.endsWithSlash
 
-    override def hashCode(): Int = {
-      var hash = segments.hashCode()
-      hash += 31 * absolute.hashCode()
-      hash += 31 * endsWithSlash.hashCode()
-      hash
-    }
+    override def hashCode(): Int =
+      // this prevents hashcode clashing when two Paths have `absolute` and `endWithSlash`
+      // asymmetric values, i.e. path1 (absolute = true, endWithSlash = false), path2 (absolute = false, endWithSlash = true)
+      {
```

Comment:
Sorry, what is "this"? Also, can we add a test case for exactly this clashing scenario you describe? Thanks! :)

Reply:
Good point on the wording - I'll rephrase. Yeah, and I'll add the example based test for the scenario described :+1: